### PR TITLE
Add a note about an open issue with code generation and SourceKit-LSP

### PR DESCRIPTION
**Pull request by t089**
Following the discussion on Slack, this adds a note for users of SourceKit-LSP informing them that the default code generation might not work as expected and offers a possible workaround.

https://swift-open-source.slack.com/archives/C02PV8T5HQD/p1726815970093299

Only updates documentation. Let me know if you feel this is the right place for this information?

### CODE REVIEW

**Comment by simonjbeaumont** [comment]
Thanks for putting this up. 

IMO I think we should move this to the _Supported platforms and minimum versions_ section, since that's where we're explaining the required runtime and development environments. This is additional note on Editor Support.

Additionally, we try to keep the README and root DocC page in sync. Would you be able to update both?

**Comment by t089** [comment]
Sorry forgot about this one, updated the PR.
CC @simonjbeaumont

File: Sources/swift-openapi-generator/Documentation.docc/Swift-OpenAPI-Generator.md [hunk 1]
```diff
@@ -107,6 +107,9 @@ The generated code, runtime library, and transports are supported on more platfo
 | Generator plugin and CLI            | ✅ 10.15+ | ✅    | ✖️      | ✖️      | ✖️       | ✖️        |
 | Generated code and runtime library  | ✅ 10.15+ | ✅    | ✅ 13+ | ✅ 13+ | ✅ 6+   | ✅ 1+    |
 
+> [!NOTE]
+> When using Visual Studio Code or other editors that rely on [SourceKit-LSP](https://github.com/swiftlang/sourcekit-lsp), the editor may not correctly recognize generated code within the same module. As a workaround, consider creating a separate target for code generation and then importing it into your main module. For more details, see the discussion in [swiftlang/sourcekit-lsp#665](https://github.com/swiftlang/sourcekit-lsp/issues/665#issuecomment-2093169169).
```
Comment (czechboy0) [inline Sources/swift-openapi-generator/Documentation.docc/Swift-OpenAPI-Generator.md:111]: The syntax in docc for notes is `> Note: ...`

Comment (t089) [reply Sources/swift-openapi-generator/Documentation.docc/Swift-OpenAPI-Generator.md]: Thanks, fixed!

**Pull request merged**